### ISSUE TITLE
The Macro-F1 metric is being calculated incorrectly.

### ISSUE THREAD

**Issue by celsofranssa** [PyTorchLightning/metrics#403]
## 🐛 Bug

F1 metric outputs wrong value when using `average="macro"`

### To Reproduce

```python
import torch
import torchmetrics
from torchmetrics import F1, MetricCollection
from sklearn.metrics import f1_score, accuracy_score

# torchmetrics
metrics = MetricCollection(
            metrics={
                "MicF1": F1(num_classes=11, average="micro"),
                "MacF1": F1(num_classes=11, average="macro")
            }
    )

# examples of target classes
target=torch.tensor([ 8, 10,  5,  2])

# examples of model predictions (LogSoftmax)
preds=torch.tensor([
        [-2.6118, -2.8115, -2.8481, -2.4452, -2.5818, -2.0106, -2.3749, -1.8872,
         -2.1726, -2.5530, -2.5767],
        [-2.5555, -2.7056, -2.8433, -2.4674, -2.5098, -1.8838, -2.3232, -2.0609,
         -2.3225, -2.4336, -2.6861],
        [-2.8012, -2.7260, -2.7958, -2.4332, -2.4809, -1.8476, -2.2616, -2.0058,
         -2.3366, -2.5866, -2.6174],
        [-2.7187, -2.7297, -2.6931, -2.2206, -2.4540, -2.1421, -2.4888, -1.9133,
         -2.0793, -2.6489, -2.7666]
    ])

# torchmetric output
metrics(torch.argmax(preds, dim=-1),target)
# {'MacF1': tensor(0.0606), 'MicF1': tensor(0.2500)}

# skitlearning output 
{
    "MicF1":f1_score(target, torch.argmax(preds, dim=-1), average='micro'),
    "MacF1": f1_score(target, torch.argmax(preds, dim=-1), average='macro')
 }
#{'MacF1': 0.13333333333333333, 'MicF1': 0.25}

```

### Expected behavior

F1 metric values (macro and micro average), should be the same as scikit-learn.

**Comment by celsofranssa** [comment]
I've found this issue is related to #295.

**Comment by SkafteNicki** [comment]
Duplicate of #295, will be fixed by #303
